PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/action/SparkRunConfigurationAction.kt

```diff
@@ -134,10 +146,13 @@ abstract class SparkRunConfigurationAction : AzureAnAction, ILogger {
                     (selectedConfigSettings.configuration as LivySparkBatchJobRunConfiguration).submitModel.mainClassName =
                             className
 
-                    runExisting(selectedConfigSettings)
+                    runExisting(selectedConfigSettings, operation)
 
                     selectedConfigSettings.isEditBeforeRun = savedIsEditBeforeRun
                 } else {
+                    EventUtil.logError(operation, ErrorType.userError, Exception("Not a runnable configuration"), null, null)
+                    operation.complete()
```

Comment:
too much exit point...

Reply:
We can have a function `EventUtil.logErrorWithComplete()`

---
In reply to: [297085702](https://github.com/microsoft/azure-tools-for-java/pull/3201#discussion_r297085702) [](ancestors = 297085702)